File: lib/domain/entities/category.dart
class CategoryEntity {
  final String id;
  final String name;
  final String? parentId;
  final String? image;
  final List<CategoryEntity> subcategories; // for tree

  CategoryEntity({
    required this.id,
    required this.name,
    this.parentId,
    this.image,
    required this.subcategories,
  });
}

File: backend/app/schemas/category.py
from pydantic import BaseModel, ConfigDict


class CategoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    slug: str
    parent_id: str | None = None
    image: str | None = None
    subcategories: list["CategoryOut"] = []


CategoryOut.model_rebuild()
